### ISSUE TITLE
cmake on vagrant yields untenable compiler flags for SSE4.2

### ISSUE THREAD

**Issue by winmillwill** [cmu-db/peloton#706]
Following these directions: https://github.com/cmu-db/peloton/wiki/Installation#build-peloton

When I invoked `make` I got:

```
[ 23%] Building CXX object src/CMakeFiles/peloton.dir/codegen/runtime_functions.cpp.o
In file included from /peloton/src/codegen/runtime_functions.cpp:15:0:
/usr/lib/gcc/x86_64-linux-gnu/4.8/include/nmmintrin.h:31:3: error: #error "SSE4.2 instruction set not enabled"
 # error "SSE4.2 instruction set not enabled"
   ^
[ 23%] Building CXX object src/CMakeFiles/peloton.dir/codegen/runtime_functions_proxy.cpp.o
[ 23%] Building CXX object src/CMakeFiles/peloton.dir/codegen/runtime_state.cpp.o
/peloton/src/codegen/runtime_functions.cpp: In static member function ‘static uint64_t peloton::codegen::RuntimeFunctions::HashCrc64(const char*, uint64_t, uint64_t)’:
/peloton/src/codegen/runtime_functions.cpp:30:37: error: ‘_mm_crc32_u64’ was not declared in this scope
       (crc) = op((crc), *(type *)buf);                   \
                                     ^
/peloton/src/codegen/runtime_functions.cpp:47:3: note: in expansion of macro ‘CRC32’
   CRC32(_mm_crc32_u64, crc, uint64_t, buf, length);
   ^
/peloton/src/codegen/runtime_functions.cpp:30:37: error: ‘_mm_crc32_u32’ was not declared in this scope
       (crc) = op((crc), *(type *)buf);                   \
                                     ^
```

Hacking the CMakeCache.txt file like so:

```
//Flags used by the compiler during all build types.
CMAKE_CXX_FLAGS:STRING=-msse4.2
```

I was able to run make without error. I haven't delved far enough into the cmake directory to figure out how to do this the right way.

**Comment by pmenon** [comment]
If `-msse4.2` is working, but `-march=native` isn't, I suspect Vagrant/VirtualBox isn't properly passing supported CPU features to GCC in the VM.  [Lo and behold, this is the case.](https://www.virtualbox.org/manual/ch09.html#sse412passthrough)

Can you try adding these lines to your Vagrant file and build again.

```
vb.customize ["setextradata", :id, "VBoxInternal/CPUM/SSE4.1", "1"]
vb.customize ["setextradata", :id, "VBoxInternal/CPUM/SSE4.2", "1"]
```

**Comment by pmenon** [comment]
Tested this out and it seems to work. Fixed in #710.

**Comment by winmillwill** [comment]
Thanks very much! Didn't have time to test before you closed the loop because of sleep and day job.
